go-controller/pkg/util/net_linux.go

```diff
@@ -0,0 +1,128 @@
+// +build linux
+
+package util
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+// GetLinkWithStateUp returns the netlink device with its state marked up
+func GetLinkWithStateUp(interfaceName string) (netlink.Link, error) {
+	link, err := netlink.LinkByName(interfaceName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup link %s: %v", interfaceName, err)
+	}
+	err = netlink.LinkSetUp(link)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set the link %s up: %v", interfaceName, err)
+	}
+	return link, nil
+}
+
+// LinkAddrAdd removes existing addresses on the link and adds the new address
+func LinkAddrAdd(link netlink.Link, address string) error {
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		return fmt.Errorf("failed to list addresses for the link %s: %v", link.Attrs().Name, err)
+	}
+	for _, addr := range addrs {
+		err = netlink.AddrDel(link, &addr)
+		if err != nil {
+			return fmt.Errorf("failed to delete address on link %s: %v", link.Attrs().Name, err)
+		}
+	}
+
+	ipnet, err := netlink.ParseIPNet(address)
+	if err != nil {
+		return fmt.Errorf("failed to parse ip %s :%v\n", address, err)
+	}
+	err = netlink.AddrAdd(link, &netlink.Addr{IPNet: ipnet})
```

Comment:
@girishmg Perhaps use AddrReplace() and then you don't have to do the list + delete dance.

Reply:
@dcbw this was more of to simulate the current `ip addr flush dev k8s-node1`.

 You can't achieve the same with `AddrReplace`

```
$ ip a add 192.171.10.3/24 dev enp12s0
$ ip a replace 192.172.10.3/24 dev enp12s0
$ ip a show dev enp12s0
5: enp12s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 50:6b:4b:fb:ec:76 brd ff:ff:ff:ff:ff:ff
    inet 192.171.10.3/24 scope global enp12s0
       valid_lft forever preferred_lft forever
    inet 192.172.10.3/24 scope global enp12s0
       valid_lft forever preferred_lft forever
```